src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/RijndaelManagedTransform.Browser.cs

```diff
@@ -0,0 +1,1307 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+
+namespace System.Security.Cryptography
+{
+    internal sealed class RijndaelManagedTransform : ICryptoTransform, ILiteSymmetricCipher
+    {
+        public const int BlockSizeBytes = 16; // 128 bits
+
+        private readonly bool _encrypting;
+
+        private int[] _encryptKeyExpansion;
+        private int[] _decryptKeyExpansion;
+
+        private readonly int _Nr;
+        private readonly int _Nb;
+        private readonly int _Nk;
+
+        private readonly int[] _encryptindex;
+        private readonly int[] _decryptindex;
+
+        private int[] _IV;
+        private int[] _lastBlockBuffer;
+        private byte[]? _depadBuffer;
+
+        public RijndaelManagedTransform(ReadOnlySpan<byte> key,
+                                        ReadOnlySpan<byte> iv,
+                                        bool encrypting)
+        {
+            _encrypting = encrypting;
+            _Nr = GetNumberOfRounds(key);
+            _Nb = BlockSizeBytes / 4;
+            _Nk = key.Length / 4;
+
+            int S1 = _Nb > 6 ? 3 : 2;
+            int S2 = _Nb > 6 ? 4 : 3;
+
+            // Precompute the modulus operations: these are performance killers when called frequently
+            int[] encryptindex1 = new int[_Nb];
+            int[] encryptindex2 = new int[_Nb];
+            int[] encryptindex3 = new int[_Nb];
+
+            int[] decryptindex1 = new int[_Nb];
+            int[] decryptindex2 = new int[_Nb];
+            int[] decryptindex3 = new int[_Nb];
```

Comment:
We can allocate _encryptIndex and _decryptIndex, and then these can just be slices over them.  Then you can remove the Array.Copy calls below, and you avoid these additional array allocations.